tests/core/test_verify_is_dir.py

```diff
@@ -0,0 +1,34 @@
+from pytest import raises
+
+from sprout.core.verify_is_dir import verify_is_dir
+
+
+def test_returns_existing_directory(tmp_path):
+    """Test that the verify_is_dir function returns the directory if it exists."""
+    assert verify_is_dir(tmp_path) == tmp_path
+
+
+def test_raises_error_with_non_existent_directory(tmp_path):
+    """Test that the verify_is_dir function raises a NotADirectoryError
+    if the directory does not exist."""
+    with raises(
+        NotADirectoryError,
+        match=r"/non_existent_directory directory doesn't exist.",
```

Comment:
Is this message required? It feels like a brittle way of running the tests since if we make a change to the function, we also need to make a change to the test here.

Reply:
You're right, it's not actually needed. We could either remove the `match` line altogether or we could only look for the "/non_existent_directory" match, like so: 
```suggestion
        match=r"/non_existent_directory",
```

Reply:
I have implemented the suggestion above and kept only the last part of the input path in `match` 👍